### PR TITLE
Fix build and clippy warnings

### PR DESCRIPTION
**Pull request by jackcmay**
Fix build and clippy warnings

Would be nice to enforce this in CI :-)

### CODE REVIEW

File: src/elf/note.rs [hunk 1]
```diff
@@ -49,10 +49,10 @@ pub const NT_PRSTATUS: u32 = 1;
 pub const NT_PRPSINFO: u32 = 3;
 
 ///Fields of siginfo_t.
-pub const NT_SIGINFO: u32 = 0x53494749;
+pub const NT_SIGINFO: u32 = 0x53_494_749;
```
Comment (jan-auer) [inline src/elf/note.rs:52]: Hex literals are probably more readable in groups of 4 (same below):

```suggestion
pub const NT_SIGINFO: u32 = 0x5349_4749;
```

Comment (jackcmay) [reply src/elf/note.rs]: @jan-auer Agreed, fixed

**Comment by jackcmay** [comment]
@m4b 👀

File: src/elf/program_header.rs [hunk 2]
```diff
@@ -314,6 +314,9 @@ macro_rules! elf_program_header_std_impl { ($size:ty) => {
                 phdrs
             }
 
+            /// # Safety
+            ///
+            /// This function creates a `ProgramHeader` directly from a raw pointer
```
Comment (m4b) [inline src/elf/program_header.rs:319]: nit: this should probably a note that the phnum value is blindly trusted and not validated in anyway (though not sure how you could)? but also maybe people should already know that if they're using this? dunno

**Pull request merged**